rotkehlchen/chain/ethereum/modules/rainbow/decoder.py

```diff
@@ -1,25 +1,5 @@
-from typing import TYPE_CHECKING
-
-from rotkehlchen.chain.evm.decoding.rainbow.constants import RAINBOW_ROUTER_CONTRACT
 from rotkehlchen.chain.evm.decoding.rainbow.decoder import RainbowCommonDecoder
 
-if TYPE_CHECKING:
-    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
-    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
-    from rotkehlchen.user_messages import MessagesAggregator
-
 
 class RainbowDecoder(RainbowCommonDecoder):
-
-    def __init__(
-            self,
-            evm_inquirer: 'EvmNodeInquirer',
-            base_tools: 'BaseDecoderTools',
-            msg_aggregator: 'MessagesAggregator',
-    ) -> None:
-        super().__init__(
-            evm_inquirer=evm_inquirer,
-            base_tools=base_tools,
-            msg_aggregator=msg_aggregator,
-            router_address=RAINBOW_ROUTER_CONTRACT,
-        )
+    ...
```

Comment:
The way you did this is wrong. You created 6 subclasses that do absolutely nothing. They are all useless.

If it's the same across all chains you need to add it as a built-in decoder here:
https://github.com/rotki/rotki/blob/c2a28b435f74a25269b4b77be678dcd5a7e3af27/rotkehlchen/chain/evm/decoding/decoder.py#L249-L268

Same as all the other decoder which are there.

If the decoder has no per-chain customization there is no reason to do all these subclasses. If you need to skip it for some chains due to whatever, add an if there same as the WETH decoder has